klient/k8s/yaml/yaml.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+)
+
+// LoadUnstructuredYAML loads a single-document YAML or JSON file
+func LoadUnstructuredYAML(filename string) (*unstructured.Unstructured, error) {
+	file, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading yaml file: %q", filename)
+	}
+	crd := unstructured.Unstructured{}
+	if err := yaml.Unmarshal(file, &crd); err != nil {
+		return nil, errors.Wrapf(err, "error decoding yaml file: %s", filename)
+	}
+	return &crd, nil
+}
+
+func listResourceFiles(directory string) ([]string, error) {
+	yamls, err := filepath.Glob(filepath.Join(directory, "*.yaml"))
+	if err != nil {
+		return nil, err
+	}
+	ymls, err := filepath.Glob(filepath.Join(directory, "*.yml"))
+	if err != nil {
+		return nil, err
+	}
+	jsons, err := filepath.Glob(filepath.Join(directory, "*.json"))
+	if err != nil {
+		return nil, err
+	}
+	return append(yamls, append(ymls, jsons...)...), nil
```

Comment:
TODO: Sort files before returning them

(currently we are returning files sorted by their extension before their name: (*.yaml, *.yml, *.json)